### PR TITLE
Es/add react nav

### PR DESCRIPTION
**Pull request by ErikSin**
# Added React Navigation

## Features included:
- Drawer Navigation (See Photo)
- Custom Header, with custom back button and menu button to open drawer (See Photo)
- Typed Screens
- Screen Components can be typed to include a static property that contains a translatable title. This title can be accessed by the parent nav, and therefore the header title can be translated
- Typed Hooks **to note: these hooks are not 100% typesafe as it relies on the developer to import proper hook into the component**

## To Do
- Add internationalization to translate titles
- Add appropriate flows/nested screens

image

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,14 +10,25 @@
...
     "expo": "~48.0.6",
     "expo-status-bar": "~1.4.4",
     "react": "18.2.0",
-    "react-native": "0.71.3"
+    "react-intl": "^6.2.10",
+    "react-native": "0.71.3",
+    "react-native-gesture-handler": "~2.9.0",
+    "react-native-reanimated": "~2.14.4",
+    "react-native-safe-area-context": "^4.5.0",
+    "react-native-screens": "^3.20.0",
```
Comment (ErikSin) [inline package.json:24]: These libraries are needed for `react-nav`

File: src/navigation/NavigationContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  ProjectInvite: NavigatorScreenParams<ProjectInviteScreens>;
+};
+
+export type SyncScreens = {
+  Bar: undefined;
+};
+
+export type ProjectInviteScreens = {
+  ProjectCoordinator: undefined;
+};
```
Comment (ErikSin) [inline src/navigation/NavigationContainer.tsx:34]: See [Type Checking With Typescript](https://reactnavigation.org/docs/typescript) in `React Nav` Docs

File: src/hooks/useTypedNavigation.ts
Comment (ErikSin) [inline src/hooks/useTypedNavigation.ts:1]: See [Annotating `useNavigation`](https://reactnavigation.org/docs/typescript#annotating-usenavigation)

**Review by achou11 — APPROVED** [review]
Looks good! Are the listed TODOs going to be part of this PR or follow-ups after merging? think it could be the latter if this PR is mostly just about nav scaffolding

**Comment by ErikSin** [comment]
> Looks good! Are the listed TODOs going to be part of this PR or follow-ups after merging? think it could be the latter if this PR is mostly just about nav scaffolding

Yeah, those to do are for another PR, i just wanted to mention it because it was related.

**Pull request merged**